util.go

```diff
@@ -17,26 +30,42 @@ func generateSigner() (ssh.Signer, error) {
 }
 
 func parsePtyRequest(s []byte) (pty Pty, ok bool) {
-	term, s, ok := parseString(s)
-	if !ok {
-		return
-	}
-	width32, s, ok := parseUint32(s)
-	if !ok {
-		return
+	reqMsg := &ptyRequestMsg{}
+	err := ssh.Unmarshal(s, reqMsg)
+	if err != nil {
+		return Pty{}, false
 	}
-	height32, _, ok := parseUint32(s)
-	if !ok {
-		return
+
+	modes := []byte(reqMsg.Modelist)
+
+	mode := struct {
+		Key uint8
+		Val uint32
+	}{}
+
+	TerminalModes := make(ssh.TerminalModes, 0)
+	for {
```

Comment:
The description on how terminal modes are actually passed can be found here: https://tools.ietf.org/html/rfc4254#section-8

Reply:
I'm not sure if I understand what you mean. Do you mean to set terminal modes to `[]byte` type?